lib/features/session/presentation/utils/session_utils.dart

```diff
@@ -0,0 +1,22 @@
+
+import 'package:core/presentation/extensions/html_extension.dart';
+import 'package:core/utils/app_logger.dart';
+
+class SessionUtils {
+  static Uri refactorApiUriForJMapDartClient(Uri apiUri) {
+    final pathSegments = apiUri.pathSegments.where((segment) => segment.isNotEmpty);
+    log('SessionUtils::refactorApiUriForJMap(): pathSegments: $pathSegments');
+    if (pathSegments.isNotEmpty && pathSegments.last == 'jmap') {
+      return apiUri;
+    } else {
+      var apiUrl = apiUri.toString();
+      if (apiUrl.endsWith('/')) {
+        apiUrl = apiUrl.append('jmap');
+      } else {
+        apiUrl = apiUrl.append('/jmap');
+      }
```

Comment:
why we need this ?

Reply:
@hoangdat 
We don't know for sure if `apiUrl` ends with the `/` character when we need to add `jmap` to the end of the path? So we need to check it.

Reply:
As I understand, `apiUrl` is the direct path to the jmap, so no need to append anything.
<img width="511" alt="image" src="https://user-images.githubusercontent.com/6462404/191735926-85123707-d467-4939-83be-779ec6fc3bf7.png">

Reply:
oke. i updated it